middle_end/flambda/inlining/inlining_report.ml

```diff
@@ -136,12 +142,22 @@ let record_decision ~dbg decision =
   end
 
 let output_then_forget_decisions ~output_prefix =
+  let l = lazy (List.rev !log) in
   if !Clflags.inlining_report then begin
     let out_channel = open_out (output_prefix ^ ".inlining.org") in
     let fmt = Format.formatter_of_out_channel out_channel in
-    Format.fprintf fmt "%a@." (print ~depth:0) (List.rev !log);
+    Format.fprintf fmt "%a@." (print ~depth:0) (Lazy.force l);
     close_out out_channel;
-    log := [];
-  end
+  end;
+  if !Clflags.inlining_report_serialized then begin
+    let ch = open_out_bin (output_prefix ^ ".inlining.obj.magic") in
```

Comment:
This filename doesn't match the help text for the command-line option (ditto for the human-readable inlining reports, above).  Also, maybe we could just use "`.inlining" for the machine-readable ones?

Reply:
I changed the extension to `.inlining` for the marshalled reports and adjusted the option descriptions to match the actual filenames generated.